### ISSUE TITLE
EnC: Adding/removing type parameters on local functions should be a rude edit

### ISSUE THREAD

**Issue by davidwengier** [dotnet/roslyn#49822]
It is currently rude edit to add/remove type parameters on a method, so for consistency we should report this rude edit on local function type parameters as well.

FYI @tmat

**Comment by tmat** [comment]
There is actually more we need to check. CLR currently does not support any edits within a generic method or type. I'm not sure if we check that for local function and if we do, do we have a test?

**Comment by davidwengier** [comment]
There are no tests that have generic local functions, but there are tests ensuring adding type parameters to methods result in a rude edit.

The following change does not result in a rude edit, which is what I thought this issue could fix at least.

From:
```cs
public class C {
    public void M() {
        void L() {
        }
    }
}
```
to:
```cs
public class C {
    public void M() {
        void L<T>() {
        }
    }
}
```

**Comment by tmat** [comment]
Does edit to a method body of an already generic local function work?

**Comment by davidwengier** [comment]
When the active statement is inside the local function, no rude edit is reported, but the edit doesn't get applied to the debugged application. If you break before the call to the local function, then the edit is applied.

**Comment by tmat** [comment]
Interesting. Could be that it works until an instantiation of the function is created or something. There could also be a difference between static local function and local function that captures variable (so it's emitted as an instance method). In any case, I think we need to report rude edit like we do with generic methods.
